crates/edr_provider/src/data.rs

```diff
@@ -2320,7 +2320,7 @@ fn create_blockchain_and_state(
 
         Ok(BlockchainAndState {
             fork_metadata: Some(ForkMetadata {
-                chain_id: blockchain.chain_id(),
+                chain_id: tokio::task::block_in_place(|| runtime.block_on(rpc_client.chain_id()))?,
```

Comment:
I suspect this can be done in parallel with the `genesis_accounts` initialization, which also blocks. Not sure if it's worth it though.